src/univariate/affine.jl

```diff
@@ -0,0 +1,222 @@
+f(x) = \\frac{1}{scale} base_dist \\! \\left( \\frac{x-shift}{scale} \\right).
+```
+
+If `base_dist` falls under a location-scale family, `affine` will attempt to return a 
+transformed distribution of type `base_dist` if an `affine` method has been implemented. 
+Otherwise, it will fall back on the `AffineDistribution` wrapper type.
+
+```julia
+affine(shift, scale, base_dist)  # location-scale transformed distribution
+params(d)            # Get the parameters, i.e. (shift, scale, base_dist)
+location(d)          # Get the location parameter
+scale(d)             # Get the scale parameter
+```
+"""
+function affine(shift::Real, scale::Real, base_dist::UnivariateDistribution)
+    T = promote_type(eltype(base_dist), typeof(shift), typeof(scale))
+    D = typeof(base_dist)
+    S = value_support(D)
+    return AffineDistribution{T,S,D}(T(shift), T(scale), base_dist)
+end
+
+# Composing affine transformations
+function affine(shift::Real, scale::Real, d::AffineDistribution)
+    return affine(shift + scale * d.shift, scale * d.scale, d.base_dist)
+end
+
+
+affine(d::UnivariateDistribution, shift::Real, scale::Real) = affine(shift, scale, d)
+
+# Aliases
+@deprecate LocationScale(args...; kwargs...) AffineDistribution(args...)
+const ContinuousAffine{T<:Real,D<:ContinuousUnivariateDistribution} = AffineDistribution{T,Continuous,D}
+const DiscreteAffine{T<:Real,D<:DiscreteUnivariateDistribution} = AffineDistribution{T,Discrete,D}
+
+
+# Support
+function minimum(d::AffineDistribution)
+    minim = d.scale > 0 ? minimum : maximum
+    return d.shift + d.scale * minim(d.base_dist)
+end
+function maximum(d::AffineDistribution)
+    maxim = d.scale > 0 ? maximum : minimum
+    return d.shift + d.scale * maxim(d.base_dist)
+end
+support(d::AffineDistribution) = affine_support(d.shift, d.scale, support(d.base_dist))
+
+affine_support(shift::Real, scale::Real, support) = shift .+ scale .* support
+function affine_support(shift::Real, scale::Real, support::RealInterval)
+    if scale > 0
+        lower = support.lb
+        upper = support.ub
+    else
+        lower = support.ub
+        upper = support.lb
+    end
+    return RealInterval(shift + scale * lower, shift + scale * upper)
+end
+
+
+#### Conversions
+
+function convert(::Type{AffineDistribution{T}}, shift::Real, scale::Real, base_dist::UnivariateDistribution) where T<:Real
+    return AffineDistribution(T(shift), T(scale), base_dist)
+end
+function convert(::Type{AffineDistribution{T}}, d::AffineDistribution{S}) where {T<:Real, S<:Real} 
+    # Should we really leave the type of d.base_dist unchanged? (Old LocationScale behavior)
+    return affine(T(d.shift), T(d.scale), d.base_dist)
+end
+
+Base.eltype(::Type{<:AffineDistribution{T,S,D}}) where {T,S,D} = promote(T, eltype(D))
+
+
+#### Parameters
+
+function location(d::AffineDistribution)
+    if hasmethod(location, (typeof(d.base_dist),))
+        return d.shift + location(d.base_dist)
+    else
+        return d.shift
+    end
+end
```

Comment:
```suggestion
location(d::AffineDistribution) = d.shift + location(d.base_dist)
```
Having a method error here is fine, something without a location doesn't get one by shifting it.

Reply:
If something doesn't have a location parameter, isn't the shift considered the location parameter? For instance, for the T distribution, I'd consider an affinely transformed version to have a location parameter equal to 0.

(This also aligns with the current convention of `LocationScale`, which is to return the shift as the location parameter; removing this behavior would be a breaking change. That being said, I consider the current behavior of treating `LocationScale(0, 1, Normal(2, 1))` as having a location of 0 to be a bug.)